### DISCUSSION TITLE
[사용자 지정] 스터디 코딩 문제 1~6번_1주차

### DISCUSSION THREAD

**Discussion by Yunda0204** [soeunl/502-3-Study#26] [General]
Q1 : 정수 num1과 num2가 매개변수로 주어집니다. 두 수가 같으면 1 다르면 -1을 retrun하도록 solution 함수를 완성해주세요.
```
int solution(int num1, int num2) {
    int answer = 0;
    return answer;
}
```
Q2 : 정수 num1, num2가 매개변수로 주어질 때, num1를 num2로 나눈 나머지를 return 하도록 solution 함수를 완성해주세요.
```
int solution(int num1, int num2) {
    int answer = -1;
    return answer;
}
```
Q3 : 머쓱이는 선생님이 몇 년도에 태어났는지 궁금해졌습니다. 2022년 기준 선생님의 나이 age가 주어질 때, 선생님의 출생 연도를 return 하는 solution 함수를 완성해주세요
```
int solution(int age) {
    int answer = 0;
    return answer;
}
```
주의) 나이가 탄생한 연도에 1살씩 매년 1월 1일마다 1살씩 증가합니다.
예) 2022년 기준 40살 1983년생입니다.

Q4 : 정수 num1과 num2가 매개변수로 주어질 때, num1을 num2로 나눈 값에 1,000을 곱한 후 정수 부분을 return 하도록 soltuion 함수를 완성해주세요.
```
int solution(int num1, int num2) {
    int answer = 0;
    return answer;
}
```
예)
num1이 3, num2가 2이므로 3 / 2 = 1.5에 1,000을 곱하면 1500이 됩니다.
num1이 7, num2가 3이므로 7 / 3 = 2.33333...에 1,000을 곱하면 2333.3333.... 이 되며, 정수 부분은 2333입니다.
num1이 1, num2가 16이므로 1 / 16 = 0.0625에 1,000을 곱하면 62.5가 되며, 정수 부분은 62입니다.

Q5 : 정수 배열 numbers가 매개변수로 주어집니다. numbers의 원소의 평균값을 return하도록 solution 함수를 완성해주세요.
```
// numbers_len은 배열 numbers의 길이입니다.
double solution(int numbers[], size_t numbers_len) {
    double answer = 0;
    return answer;
}
```
주의) 정답의 소수 부분이 .0 또는 .5인 경우만 입력으로 주어집니다.
예)
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10] = 5.5

**Q6 : 다음과 같은 홀짝 게임이 있습니다. 
게임을 시작할 때 소지금액은 200,000원이며, 목표금액은 100,000,000원입니다. 
한 번 승리할 때마다 소지금액이 2배가 됩니다. 
이 때, 승률이 50%인 경우 몇 번의 연승이 필요한지, 
그리고 필요한 연승을 달성할 확률과 최종 소지금액은 얼마인지 계산하세요.**

**Comment by heyejiyang** [comment]
Q1.
```
class Solution {
    public int solution(int num1, int num2) {
        return (num1==num2) ? 1 : -1;
    }
}
```

Q2.
```
class Solution {
    public int solution(int num1, int num2) {
        return num1 % num2;
    }
}
```
Q3.
```
class Solution {
    public int solution(int age) {
        int answer = 2022 - (age-1);
        return answer;
    }
}
```
Q4.
```
class Solution {
    public int solution(int num1, int num2) {

        return (int)((double)num1/num2 * 1000);
    }
}
```
Q5.
```
class Solution {
    public double solution(int[] numbers) {
        double sum = 0.0;
       for(int i = 0; i< numbers.length; i ++){
           sum += numbers[i];
       }
        double answer = sum / numbers.length;
        return answer;
    }
}
```

**Comment by hyeon12** [comment]
Q1.
```
    int solution(int num1, int num2) {
        return (num1 == num2)? 1 : -1;
    }
```

Q2.
```
    int solution(int num1, int num2) {
        int answer = num1 % num2;
        return answer;
    }
```

Q3.
```
    int solution(int age) {
        int answer = 2022-(age-1);
        return answer;
    }
```

Q4.
```
    int solution(int num1, int num2) {
        int answer = (int)((double)num1/num2 *1000);
        return answer;
    }
```

Q5.
```
    double solution(int numbers[]) {
        double total = 0;
        for (int i = 0; i < numbers.length; i++) {
            total += numbers[i];
        }
        double answer = total/numbers.length;
        return answer;
    }
```

Q6.
```
public class Game {
    public static void main(String[] args) {
        int wins = 0; //연승 횟수

        int money = 200000, goal = 100000000;
        while(money <= goal){
            wins++;
            money *= 2;
        }

        double continuity = 1.0;
        for (int i = 0; i < wins; i++) {
            continuity *= 0.5;
        }


        //double continuity = Math.pow(0.5, wins);

        System.out.printf("연승확률은 %f, 최종 소지금액은 %d 입니다.%n", continuity, money);
    }
}
```

**Comment by soeunl** [comment]
![KakaoTalk_20240506_163520685](https://github.com/soeunl/502-3-Study/assets/154653812/b0cd5efe-ad80-4d6d-bcfc-11a73511f834)
![KakaoTalk_20240506_163520685_04](https://github.com/soeunl/502-3-Study/assets/154653812/e7eaa00f-15b6-4854-af35-3cc66d0f0f0b)
![KakaoTalk_20240506_163520685_03](https://github.com/soeunl/502-3-Study/assets/154653812/4db21871-e75f-4bc7-9f46-e15744b4a507)
![KakaoTalk_20240506_163520685_02](https://github.com/soeunl/502-3-Study/assets/154653812/a96d46e1-57e4-472f-b0ca-9a7ea74ab869)
![KakaoTalk_20240506_163520685_01](https://github.com/soeunl/502-3-Study/assets/154653812/afa5d81f-c43e-4500-a815-1e3160ab5548)

**Reply by siwon9** [reply]
```
package QuestionW1;

public class Solution_so {
    public static double solution(int num1, int num2) {
        return (double)num1 / num2 * 1000;
    }

    public static void main(String[] args) {
        int answer = (int)solution(3,2);
        System.out.println("결과는?" + answer);
    }
}
```

**Comment by beom33** [comment]
Q1.

           class solution
             int solution(int num1, int num2) {
               if (num1 == num2) {
             return 1;
                } else {
             return -1;
           }  
         } 




Q2.
        

               class solution  {
           int solution(int num1, int num2) {
               return  num1/num2 ;

               }
          }


Q3.

             class solution {
             int solution(int age) {
                 int answer = 2022 - (age+1); 
                  return answer;
              }
        }
     


Q.4

         int solution(int num1, int num2) {
              int answer = (num1 / num2) * 1000;
                   return answer;
            }
 


Q5.

**Comment by Yunda0204** [comment]
Q6 
20만원으로 시작한 홀짝게임으로 1억을 벌려면
몇연승 필요한지, 연승의 확률, 최종소지금액을 구하시오.
ps. 1승시 2배, 복리로 배팅할것.

    public static void main(String[] args) {
        int Wins = 0; // 연승횟수
        int totalMoney = 200000; // 시작 금액
        int targetMoney = 100000000; // 목표금액`

        while (totalMoney < targetMoney) {
            Wins++; // 소지금액이 목표보다 작을때마다 1승 추가
            totalMoney *= 2; // 그때 소지금액 *2
        }

        // 승률 50퍼 * 필요연승 = 9연승 확률
        double continuity = Math.pow(0.5, Wins);


        System.out.println("필요 연승 : " + Wins);
        System.out.println("연승 확률 : " + continuity);
        System.out.println("달성 금액 : " + totalMoney);
    }

**Reply by Yunda0204** [reply]
연승확률에 * 100을 할 필요는 없었네요

**Reply by Yunda0204** [reply]
```
   // 승률 50% * 50% * 50% ... = * 100
    double continuity = Math.pow(0.5, Wins) * 100;
 ```
    
해야돼요

**Reply by Yunda0204** [reply]
순수하게 연승확률 구하는거고 
진지하게 파고들면 매판 승률 50%니깐 
생각보다 할만한 확률입니다.
독립시행이니깐요
